airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/CdcMysqlSourceTest.java

```diff
@@ -0,0 +1,446 @@
+      String dateTimeValue = c.get("datetime_col").asText();
+      ((ObjectNode) c).put("datetime_col", dateTimeValue.substring(1, dateTimeValue.length() - 1));
+    }).collect(Collectors.toSet());
+
+    AirbyteCatalog discover = source.discover(config);
+    List<AirbyteStream> streams = discover.getStreams();
+
+    assertEquals(streams.size(), 1);
+    JsonNode jsonSchema = streams.get(0).getJsonSchema().get("properties");
+    assertEquals(jsonSchema.get("id").get("type").asText(), "number");
+    assertEquals(jsonSchema.get("datetime_col").get("type").asText(), "string");
+
+    AirbyteCatalog catalog = new AirbyteCatalog().withStreams(streams);
+    final ConfiguredAirbyteCatalog configuredCatalog = CatalogHelpers.toDefaultConfiguredCatalog(catalog);
+
+    configuredCatalog.getStreams().forEach(c -> c.setSyncMode(SyncMode.INCREMENTAL));
+    Set<JsonNode> dataFromDebeziumSnapshot =
+        extractRecordMessages(AutoCloseableIterators.toListAndClose(source.read(config, configuredCatalog, null)))
+            .stream()
+            .map(airbyteRecordMessage -> {
+              JsonNode data = airbyteRecordMessage.getData();
+              removeCDCColumns((ObjectNode) data);
+              return data;
+            })
+            .collect(Collectors.toSet());
+
+    assertEquals(originalData, dataFromDebeziumSnapshot);
+
+    // TODO: Fix full refresh (non-cdc) mode. The value of the datetime_col is adjusted by the TIMEZONE
+    // the code is running in,
+    // in my case it got adjusted to IST i.e. "2013-09-05T15:40:02Z" and "2013-09-06T15:40:02Z".
+    // configuredCatalog.getStreams().forEach(c -> c.setSyncMode(SyncMode.FULL_REFRESH));
+    // Set<JsonNode> dataFromFullRefresh = extractRecordMessages(
+    // AutoCloseableIterators.toListAndClose(source.read(config, configuredCatalog, null)))
+    // .stream()
+    // .map(AirbyteRecordMessage::getData).collect(Collectors.toSet());
+    // assertEquals(dataFromFullRefresh, originalData);
+  }
+
+  private void setupForDateTimeDataTypeTest(ImmutableList<JsonNode> data) {
+    executeQuery("CREATE DATABASE " + "test_schema" + ";");
+    executeQuery(String.format(
+        "CREATE TABLE %s.%s(%s INTEGER, %s DATETIME, PRIMARY KEY (%s));",
+        "test_schema", "table_with_date_time", "id", "datetime_col", "id"));
+
+    executeQuery(String
+        .format("INSERT INTO %s.%s (%s, %s) VALUES (%s, %s);", "test_schema",
+            "table_with_date_time",
+            "id", "datetime_col",
+            data.get(0).get("id").asInt(), data.get(0).get("datetime_col").asText()));
+
+    executeQuery(String
+        .format("INSERT INTO %s.%s (%s, %s) VALUES (%s, %s);", "test_schema",
+            "table_with_date_time",
+            "id", "datetime_col",
+            data.get(1).get("id").asInt(), data.get(1).get("datetime_col").asText()));
+    ((ObjectNode) config).put("database", "test_schema");
+  }
+
+  @Test
+  protected void syncShouldHandlePurgedLogsGracefully() throws Exception {
```

Comment:
can this live in the base cdc source test? (and sources could explicitly opt out if it's not supported)